sdk/storage/Azure.Storage.DataMovement/src/Shared/DataTransfer.cs

```diff
@@ -77,37 +77,21 @@ public async ValueTask DisposeAsync()
         /// </summary>
         public void EnsureCompleted(CancellationToken cancellationToken = default)
         {
-#if DEBUG
-            VerifyTaskCompleted(HasCompleted);
-#endif
 #pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
             AwaitCompletion(cancellationToken);
 #pragma warning restore AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
         }
 
-        [Conditional("DEBUG")]
-        private static void VerifyTaskCompleted(bool isCompleted)
-        {
-            if (!isCompleted)
-            {
-                if (Debugger.IsAttached)
-                {
-                    Debugger.Break();
-                }
-            }
-        }
-
         /// <summary>
         /// Waits until the data transfer itself has completed
         /// </summary>
         /// <param name="cancellationToken"></param>
         public Task AwaitCompletion(CancellationToken cancellationToken = default)
         {
+            SpinWait spinWait = new SpinWait();
             while (!HasCompleted)
             {
-#if DEBUG
-                VerifyTaskCompleted(HasCompleted);
-#endif
+                spinWait.SpinOnce();
```

Comment:
@schaabs Definitely looking for something better if you're open to a discussion about.

We cant await a task since multiple tasks can be used to complete one DataTransfer.